File: backend/src/main/java/com/manabihub/ai/service/AiChatService.java
package com.manabihub.ai.service;

import com.manabihub.ai.dto.request.AiChatMessageRequest;
import com.manabihub.ai.dto.response.AiChatEligibilityResponse;
import com.manabihub.ai.dto.response.AiChatMessageResponse;

import java.util.UUID;

public interface AiChatService {

    AiChatEligibilityResponse getEligibility(UUID courseId, UUID lessonBlockId);

    AiChatMessageResponse sendMessage(UUID courseId, UUID lessonBlockId, AiChatMessageRequest request);
}

File: frontend/src/features/ai-chat/services/aiChatService.ts
import { axiosClient } from '../../../shared/api/axiosClient';
import { ENDPOINTS } from '../../../shared/api/endpoints';
import type { ApiResponse } from '../../../shared/types/api';
import type { AiChatEligibility, AiChatMessage, AiChatMessageRequest } from '../types';

export const aiChatService = {
  getEligibility: async (courseId: string, lessonBlockId: string): Promise<AiChatEligibility> => {
    const response = await axiosClient.get<ApiResponse<AiChatEligibility>>(
      ENDPOINTS.studentAiChat.eligibility(courseId, lessonBlockId),
    );
    return response.data.data;
  },

  sendMessage: async (
    courseId: string,
    lessonBlockId: string,
    request: AiChatMessageRequest,
  ): Promise<AiChatMessage> => {
    const response = await axiosClient.post<ApiResponse<AiChatMessage>>(
      ENDPOINTS.studentAiChat.messages(courseId, lessonBlockId),
      request,
    );
    return response.data.data;
  },
};
